app/src/main/java/android/example/courses/NavigationActivity.kt

```diff
@@ -31,14 +39,51 @@ class NavigationActivity : AppCompatActivity(),
 
         val navController = host.navController
 
-        val drawerLayout : DrawerLayout? = findViewById(R.id.drawer_layout)
+        val drawerLayout: DrawerLayout? = findViewById(R.id.drawer_layout)
         appBarConfiguration = AppBarConfiguration(
-            setOf(R.id.home_dest, R.id.courses_dest, R.id.schedule_dest),
+            setOf(
+                R.id.home_dest,
+                R.id.courses_dest,
+                R.id.schedule_dest,
+                R.id.attendance_dest,
+                R.id.grades_dest,
+                R.id.documents_dest
+            ),
             drawerLayout)
 
         setupActionBar(navController, appBarConfiguration)
-
         setupNavigationMenu(navController)
+        setupNavHeader()
+    }
+
+    private fun setupNavHeader() {
+        auth = FirebaseAuth.getInstance()
+
+        val navView = findViewById<NavigationView>(R.id.nav_view)
+        val headerView = navView.inflateHeaderView(R.layout.side_header_layout)
+
+        val nameView = headerView.findViewById<TextView>(R.id.headerNameTextView)
+        val photoView = headerView.findViewById<CircularImageView>(R.id.profileImage)
+
+        var email = auth.currentUser?.email!!
+
+        db.collection("users").document(email)
+            .get()
+            .addOnSuccessListener {document ->
+                if (document != null) {
+                    nameView.text = (document.data?.get("name") as String)
+
+                    val link = (document.data?.get("pictureUrl") as String)
+                    Picasso.get()
```

Comment:
I saw that on a slow network the profile picture is downloaded in a few seconds, during which is not displayed at all. How can I fix that? Should I put a loading screen right after the login step? If so, how can I know that all the data was loaded completely to be able to hide it?